src/app-layout/__tests__/desktop.test.tsx

```diff
@@ -235,8 +233,14 @@ describeEachAppLayout({ sizes: ['desktop'] }, ({ theme }) => {
   test('should render badge when defined', () => {
     const { wrapper } = renderComponent(<AppLayout drawers={manyDrawers} />);
 
-    expect(isDrawerTriggerWithBadge(wrapper, manyDrawers[0].id)).toEqual(true);
-    expect(isDrawerTriggerWithBadge(wrapper, manyDrawers[1].id)).toEqual(false);
+    expect(wrapper.findDrawerTriggerById(manyDrawers[0].id, { hasBadge: true })).toBeTruthy();
+    expect(wrapper.findDrawerTriggerById(manyDrawers[1].id, { hasBadge: false })).toBeTruthy();
+  });
+
+  test('should return null when finding a non-existing drawer with hasBadge condition', () => {
+    const { wrapper } = renderComponent(<AppLayout drawers={manyDrawers} />);
+
+    expect(wrapper.findDrawerTriggerById('non-existing', { hasBadge: true })).toBeNull();
```

Comment:
nit: should we add another line and ensure it is null when hasBase is false?

Reply:
Codecov was okay without this check, but fine, I added it